sdk/storage/Azure.Storage.Common/tests/Shared/StorageTestBase.cs

```diff
@@ -61,18 +62,28 @@ public void StopLoggingEvents()
         }
 
         /// <summary>
-        /// Output the Events to the console in the case of failure.
+        /// Sets up the Event listener buffer for the test about to run.
+        /// </summary>
+        [SetUp]
+        public void SetupEventsForTest()
+        {
+            if (s_listener != null)
+            {
+                TestEventListener.Events = new StringBuilder();
```

Comment:
I'd rather isolate this kind of functionality inside the `TestEventListener` class rather than spread the implementation.

Reply:
We need to at least flush the buffer at the end of each test. I can move this to the teardown within StorageTestBase but figured this would be more straight forward.

Reply:
I don't think spreading implementation state changes across different classes is as straightforward as `StartTest` and `StopTest` methods on `TestEventListener`.

Reply:
I was assuming that the Setup/TearDown methods would need to be on the Test class.
I meant that having a separate Setup method was more straightforward than putting everything on just the TearDown method.
I can try putting it on the TestEventListener.

Reply:
I tried this out and it doesn't work. It looks like the Setup method needs to be on a class in the test class hierarchy:
https://nunit.org/docs/2.2.7/setup.html

Reply:
Though I can encapsulate the state in methods within TestEventListener (which is what you were suggesting all along).